tests/tools.vitruv.applications.pcmumlclass.tests/src/tools/vitruv/applications/pcmumlclass/tests/MediaStoreRepositoryCreationTest.xtend

```diff
@@ -165,4 +175,331 @@ class MediaStoreRepositoryCreationTest extends LegacyPcmUmlClassApplicationTest
+				return
+			}
+
+			val originalInterface = originalContractsPackageElement as Interface
+			val insertedInterface = EcoreUtil.copy(originalInterface)
+
+			// add each operation without its parameters because at least the returnParameters will be already generated
+			for (operation : insertedInterface.ownedOperations) {
+				operation.ownedParameters.clear
+			}
+			contractsPackage.packagedElements += insertedInterface
+		]
+
+		if (!(originalContractsPackageElement instanceof Interface)) {
+			return
+		}
+
+		changeUmlView[
+			val repositoryPackage = UmlQueryUtil.claimUmlModel(it, modelName).nestedPackages.head
+			assertNotNull(repositoryPackage)
+			var contractsPackage = repositoryPackage.nestedPackages.findFirst [
+				it.name == DefaultLiterals.CONTRACTS_PACKAGE_NAME
+			]
+			assertNotNull(contractsPackage)
+			val originalInterface = originalContractsPackageElement as Interface
+			var generatedInterface = contractsPackage.packagedElements.filter(Interface).findFirst [
+				it.name == originalInterface.name
+			]
+			assertNotNull(generatedInterface)
+
+			for (originalOperation : originalInterface.operations) {
+				var generatedOperation = generatedInterface.operations.findFirst[it.name == originalOperation.name]
+
+				// remove already generated parameters
+				generatedOperation.ownedParameters.clear
+				for (originalParameter : originalOperation.ownedParameters) {
+					val insertedParameter = EcoreUtil.copy(originalParameter)
+					insertedParameter.type = originalParameter.type === null
+						? null
+						: UmlQueryUtil.claimUmlModel(it, modelName).eAllContents.toList.filter[it instanceof Type].map [
+						it as Type
+					].findFirst [
+						it.name == originalParameter.type.name
+					]
+					generatedOperation.ownedParameters += insertedParameter
+				}
+			}
+		]
+	}
+
+	private def simulateComponentInsertion_UML(String modelName, Package originalComponentPackage,
+		int userDisambigutationComponentType) {
+		val originalComponentImpl = originalComponentPackage.packagedElements.filter(Class).head
+		assertNotNull(originalComponentImpl)
+
+		// should generate generatedComponentImpl
+		changeUmlView [
+			var generatedModel = UmlQueryUtil.claimUmlModel(it, modelName)
+			var generatedRepositoryPackage = generatedModel.nestedPackages.head
+			assertNotNull(generatedRepositoryPackage)
+
+			// Simulate adding a Component via round-trip by adding the components package.
+			// For that, first remove the implementation from the original package to avoid duplication.
+			originalComponentPackage.packagedElements -= originalComponentImpl
+			assertTrue(originalComponentPackage.packagedElements.empty)
+
+			userInteraction.addNextSingleSelection(userDisambigutationComponentType)
+			generatedRepositoryPackage.nestedPackages += originalComponentPackage // throws UUID error when tested on its own
+		]
+
+		// should generate the constructor Parameters corresponding to RequiredRoles
```

Comment:
"should" sounds like the intent of the method part is unclear — Can be rephrased to avoid that.